app/packages/core/src/components/applications/Applications.tsx

```diff
@@ -0,0 +1,207 @@
+import { HubOutlined, InsightsOutlined, PeopleOutlined } from '@mui/icons-material';
+import { Box, Button, Chip, Divider, IconButton, List, ListItem, ListItemText, Typography } from '@mui/material';
+import { useQuery } from '@tanstack/react-query';
+import { Fragment, FunctionComponent, useContext, useState } from 'react';
+import { Link } from 'react-router-dom';
+
+import ApplicationInsights from './ApplicationsInsights';
+import { IApplicationOptions } from './utils';
+
+import { APIContext, APIError, IAPIContext } from '../../context/APIContext';
+import { IApplication } from '../../crds/application';
+import Pagination from '../utils/Pagination';
+import UseQueryWrapper from '../utils/UseQueryWrapper';
+
+interface IApplicationProps {
+  application: IApplication;
+}
+
+/**
+ * The `Application` component renders a single application in the list of applications. Within the list we show the
+ * name, description, tags, teams and dependencies of an application. If the application has insights we also render a
+ * button to show the insights in a drawer via the `ApplicationInsights` component.
+ */
+const Application: FunctionComponent<IApplicationProps> = ({ application }) => {
+  const [open, setOpen] = useState(false);
+
+  /**
+   * `hideInsights` sets the `open` state to `false`, which will hide the application insights drawer.
+   */
+  const hideInsights = () => {
+    setOpen(false);
+  };
+
+  /**
+   * `showInsights` sets the `open` state to `true`, which will show the application insights drawer. We have to call
+   * `e.preventDefault()` because the button which uses this function is rendered inside a React Router `Link` and
+   * without this the link would be triggered.
+   */
+  const showInsights = (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
+    e.preventDefault();
+    setOpen(true);
+  };
+
+  return (
+    <>
+      <ListItem
+        component={Link}
+        to={`/applications${application.id}`}
+        sx={{ color: 'inherit', textDecoration: 'inherit' }}
+        secondaryAction={
+          application.insights && application.insights.length > 0 ? (
+            <IconButton edge="end" onClick={showInsights}>
+              <InsightsOutlined />
```

Comment:
The icon is to close to the right edge, add some horizontal padding to fix

Reply:
This is the default by MUI, not sure if we should change this.